src/compile/data/optimize.ts

```diff
@@ -114,6 +115,33 @@ function getLeaves(roots: DataFlowNode[]) {
   return leaves;
 }
 
+export function isParseMergeable(x: ParseNode, y: ParseNode) {
```

Comment:
Add comment explaining what's the criteria for something to be mergeable?  It seems like that the intersection of the two should have identical values.